### PR TITLE
IG Manager: return the instance group name once even if it existis in…

### PR DESCRIPTION
**Pull request by mmamczur**
… many nodes.

The manager sync function assumes the names of the groups are listed without a zone and will attempt to do the add/remove operations in all zones. Before this change, if the group existed in more than one zone it would be returned many times (list will contain the same name a few times). This leads to unnecessary iterations and instances listing - everything would happen N^2 for N=number of zones.

### CODE REVIEW

File: pkg/controller/controller_test.go [hunk 1]
```diff
@@ -675,7 +675,7 @@ func TestMCIngressIG(t *testing.T) {
 	if err != nil {
 		t.Errorf("lbc.instancePool.List() = _, %v, want nil", err)
 	}
-	var wantInstanceGroups []string
+	wantInstanceGroups := []string{}
```
Comment (code-elinka) [inline pkg/controller/controller_test.go:678]: why did you change it?

File: pkg/instancegroups/manager.go [hunk 2]
```diff
@@ -226,6 +226,8 @@ func (m *manager) Get(name, zone string) (*compute.InstanceGroup, error) {
 }
 
 // List lists the names of all Instance Groups belonging to this cluster.
+// It will return only the names of the groups without the zone. If w group with the same name exists in more than one
```
Comment (code-elinka) [inline pkg/instancegroups/manager.go:229]:
```suggestion
// It will return only the names of the groups without the zone. If a group with the same name exists in more than one
```

**Comment by code-elinka** [comment]
Looks good

**Comment by code-elinka** [comment]
nit: it is better to have a short commit message header 
Screenshot from 2024-01-16 10-04-40
Screenshot from 2024-01-16 10-04-34

see guidelines for commit messages: https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

**Comment by mmamczur** [comment]
> nit: it is better to have a short commit message header

I changed it (but only in the commit message and not PR title)
